bootstrap.py

```diff
@@ -32,4 +29,82 @@ def main(coderoot: str, argv: Sequence[str] | None = None) -> int:
         print(f"repo {args.repo} not supported yet!")
         return 1
 
+    # xcode-select --install will take a while,
+    # and involves elevated permissions with a GUI,
+    # so best to just let the user go through that separately then retrying,
+    # rather than waiting for it.
+    # There is a way to perform a headless install but it's more complex
+    # (refer to how homebrew does it).
+    try:
+        xcode_git = proc.run(("/usr/bin/xcrun", "-f", "git"))
```

Comment:
Can you break this up into functions to avoid one large main block

Reply:
This is pretty much as good as it gets... it looks largeish but it's mainly that big block of text and a few long subprocess argument lists that get spread out. It's really not that large in terms of cyclomatic complexity